src/main/java/ch/njol/skript/expressions/ExprListIndex.java

```diff
@@ -0,0 +1,119 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.Variable;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.LiteralUtils;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+@Name("List Index")
+@Description("An expression to obtain the first, last, or all occurrences of something in a list.")
+@Examples({
+	"command /listindex:",
+	"\ttrigger:",
+	"\t\tset {_list::*} to 1,2,3,4,5,1",
+	"\t\tsend \"%first index of 1 in %{_list::*}%%\"",
+	"\t\tsend \"%last index of 1 in %{_list::*}%%\"",
+	"\t\tsend \"%all indexes of 1 in %{_list::*}%%\""
+})
+public class ExprListIndex extends SimpleExpression<Object> {
+	static {
+		Skript.registerExpression(ExprListIndex.class, Object.class, ExpressionType.COMBINED,
+			"[(first|last)] index of %object% in %objects%",
+			"[all] ind(ex|ic)es of %object% in %objects%");
+	}
+
+	private Expression<?> elementExpr, listExpr;
+
+	private boolean variable, all, first;
+
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		variable = (parseResult.mark & 1) == 1;
+		all = matchedPattern == 1;
+		first = (parseResult.mark & 2) == 0;
+		elementExpr = LiteralUtils.defendExpression(exprs[0]);
+		listExpr = LiteralUtils.defendExpression(exprs[1]);
+		if (listExpr.isSingle())
+			return false;
+		if (variable && !(listExpr instanceof Variable)) {
+			Skript.error(listExpr + " is not a list variable");
+			return false;
+		}
+		return LiteralUtils.canInitSafely(elementExpr, listExpr);
+	}
+
+	@Override
+	@Nullable
+	protected Object[] get(Event e) {
```

Comment:
Change all `e` to `event`
```suggestion
	protected Object[] get(Event event) {
```